hammer/sim/xcelium/__init__.py

```diff
@@ -377,6 +381,336 @@ def generate_sim_tcl(self) -> bool:
     f.close()  
     return True
 
+  def generate_amscf(self) -> bool:
+    # Open AMS control file template for read.
+    # Hardcoded path for now
+    t = open("amscf_template.scs", "r")
+
+    # Create AMS control file (or overwrite if one already exists) for read/write.
+    # Hardcoded path for now.
+    f = open(f"./src/amscf.scs", "w+")
+
+    # Get absolute paths for analog models from PDK and schematics from extralibs, respectively.
+    model_path = self.get_setting("sim.xcelium.anamodels")
+    models = [modelfile for modelfile in os.scandir(model_path)]
+    schematic_path = self.get_setting("sim.xcelium.schematics")
+    schematics = [schematic for schematic in os.scandir(schematic_path)]
+
+    # Get list of paths to individual files within the PDK models (?) and schematic directories, respectively.
+    #models = []
+    #schematics = []
+
+    # Warnings for missing files.
+    if (len(schematics) > 0 and len(models) == 0):
+      self.logger.warning(f"No models found in {model_path} to support analog schematics.")
+    else:
+      if (len(models) == 0):
+        self.logger.warning(f"No models found in {model_path}.")
+      if (len(schematicpath) == 0):
+        self.logger.warning(f"No analog schematics found {schematics}.")
+
+    # Get string representation of AMS control file template.
+    template = t.read()
+
+    # Format modelpaths list as a single string with include statements.
+    formatted_models = ""
+    for modelpath in models:
+      formatted_models += f"include {modelpath}"
+    
+    # Format schematicpaths list as a single string with include statements.
+    formatted_schematics = ""
+    for schematicpath in schematics:
+      formatted_schematics += f"include {schematicpath}"
+
+    # Replace empty model deck with formatted string of model paths.
+    template = re.sub("// model deck\n", "// model deck\n" + formatted_models)
+
+    # Replace empty schematic deck with formatted string of schematic paths.
+    template = re.sub("// schematic deck\n", "// schematic deck\n" + formatted_schematics)
+
+    # Write filled template to AMS control file.
+    f.write(template)
+
+    # Close files properly.
+    t.close()
+    f.close()
+    return True
+  
+  def retrieve_files(path, exts=[], output_type=str, relative=True):
```

Comment:
This seems like a generally-useful utility. Consider moving it to `hammer/utils/__init__.py`?